metals/src/main/scala/scala/meta/internal/metals/GithubNewIssueUrlCreator.scala

```diff
@@ -0,0 +1,96 @@
+package scala.meta.internal.metals
+
+import java.net.URLEncoder
+
+import scala.util.Properties
+
+import scala.meta.internal.bsp.BspResolvedResult
+import scala.meta.internal.bsp.BspSession
+import scala.meta.internal.bsp.ResolvedBloop
+import scala.meta.internal.bsp.ResolvedBspOne
+import scala.meta.internal.builds.BuildTools
+
+import org.eclipse.lsp4j.ClientInfo
+
+class GithubNewIssueUrlCreator(
+    tables: Tables,
+    buildTargets: BuildTargets,
+    currentBuildServer: () => Option[BspSession],
+    calculateNewBuildServer: () => BspResolvedResult,
+    clientInfo: ClientInfo,
+    buildTools: BuildTools,
+) {
+
+  def buildUrl(): String = {
+    val scalaVersions =
+      buildTargets.allScala.map(_.scalaVersion).toSet.mkString("; ")
+    val clientVersion =
+      Option(clientInfo.getVersion()).map(v => s" v$v").getOrElse("")
+    val body =
+      s"""|<!--
+          |        Describe the bug ...
+          |
+          |        Reproduction steps
+          |          1. Go to ...
+          |          2. Click on ...
+          |          3. Scroll down to ...
+          |          4. See error
+          |-->
+          |
+          |### Expected behaviour:
+          |
+          |<!-- A clear and concise description of what you expected to happen. -->
+          |
+          |**Operating system:**
+          |${Properties.osName}
+          |
+          |**Java version:**
+          |${Properties.javaVersion}
+          |
+          |**Editor/extension:**
+          |${clientInfo.getName()}$clientVersion
+          |
+          |**Metals version:**
+          |${BuildInfo.metalsVersion}
+          |
+          |### Extra context or search terms:
+          |<!--
+          |        - Any other context about the problem
+          |        - Search terms to help others discover this
+          |-->
+          |
+          |### Workspace information:
+          |
+          | - **Scala versions:** $scalaVersions$selectedBuildTool$selectedBuildServer
+          | - **All build tools in workspace:** ${buildTools.all.mkString("; ")}
+          |""".stripMargin
+    s"https://github.com/scalameta/metals/issues/new?body=${URLEncoder.encode(body)}"
+  }
+
+  private def selectedBuildTool(): String = {
+    tables.buildTool
+      .selectedBuildTool()
+      .map { value =>
+        s"""|
+            | - **Build tool:** ${value}""".stripMargin
+      }
+      .getOrElse("")
+  }
+
+  private def selectedBuildServer(): String = {
+    currentBuildServer()
+      .map(s => s"${s.main.name} v${s.main.version}")
+      .orElse {
+        calculateNewBuildServer() match {
```

Comment:
I wonder if this is not an actual data point here, since if you need to calcualte it then it's not connected. Maybe you could make the name soemthing like `Disconnected: Bloop` etc. ?